File: src/ProjectR.Scripting/content/scripts/afflictions/Passives/lua/Boss.lua
function GetName()
  return "Boss"
end

function GetType()
  return Passive
end

function GetHookPoints()
  return { UsingMP }
end

function OnAttachment(char)
  local stats = GetStats(char)
  local hpStat = stats:GetSingleStat(HP)
  local baseHP = hpStat:Get(Base)
  baseHP = baseHP * 20
  hpStat:Set(Base, baseHP)

  for i = AD, MR do
    local stat = stats:GetSingleStat(i)
    local baseStat = stat:Get(Base)
    baseStat = baseStat * 0.8
    stat:Set(Base, baseStat)
  end

  local dthStat = stats:GetSingleStat(DTH)
  local baseDTH = dthStat:Get(Base)
  baseDTH = baseDTH + 500
  dthStat:Set(Base, baseDTH)

  char:Heal(char:GetMaxHP())
end

function OnRemoval(char)
  local stats = GetStats(char)
  local hpStat = stats:GetSingleStat(HP)
  local baseHP = hpStat:Get(Base)
  baseHP = baseHP / 20
  hpStat:Set(Base, baseHP)

  for i = AD, MR do
    local stat = stats:GetSingleStat(i)
    local baseStat = stat:Get(Base)
    baseStat = baseStat / 0.8
    stat:Set(Base, baseStat)
  end

  local dthStat = stats:GetSingleStat(DTH)
  local baseDTH = dthStat:Get(Base)
  baseDTH = baseDTH - 500
  dthStat:Set(Base, baseDTH)
end

function Trigger(hookPoint, char)
  if hookPoint == UsingMP then
    param1 = 0
  end
end

File: src/ProjectR.Scripting/content/scripts/afflictions/Passives/Boss.cs
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting.Afflictions
{
    public class Boss : Affliction
    {
        public override string Name { get { return "Boss"; } }

        public override AfflictionType Type { get { return AfflictionType.Passive; } }

        protected override HookPoint[] HookPoints
        {
            get
            {
                return new[]
                {
                    HookPoint.UsingMP, 
                };
            }
        }

        protected override void OnAttachment(ICharacter character)
        {
            var stats = character.Stats;
            stats[BaseStat.HP][StatType.Base] *= 20;

            for (var i = BaseStat.AD; i <= BaseStat.MR; ++i)
            {
                stats[i][StatType.Base] *= .8;
            }

            stats[DebuffResistance.DTH][StatType.Base] += 500;

            character.Heal(stats.GetTotalStat(BaseStat.HP));
        }

        protected override void OnRemoval(ICharacter character)
        {
            var stats = character.Stats;
            stats[BaseStat.HP][StatType.Base] /= 20;

            for (var i = BaseStat.AD; i <= BaseStat.MR; ++i)
            {
                stats[i][StatType.Base] /= .8;
            }

            stats[DebuffResistance.DTH][StatType.Base] -= 500;

            character.Heal(stats.GetTotalStat(BaseStat.HP));
        }

        protected override void OnUsingMP(ICharacter character, ref double value)
        {
            value = 0;
        }
    }
}